tools/wt_verify/wt_verify.py

```diff
@@ -172,6 +243,151 @@ def parse_output():
     return output
 
 
+def show_block_distribution_broken_barh(filename, checkpoints, bar_width=1):
+    """
+    Make a "broken" horizontal bar plot to show the on disk information contained in a file.
+    checkpoints: dictionary generated by parse_dump_blocks() that contains information related to
+    different checkpoints.
+    """
+    bar_width = 1
+    colors = ['blue', 'orange', 'green', 'red', 'purple', 'brown', 'pink', 'gray', 'olive', 'cyan']
+
+    # Remember the max offset seen and the associated data size.
+    max_addr = 0
+    max_addr_size = 0
+
+    color_indx = 0
+    page_types = []
+    fig, ax = plt.subplots(1, figsize=(15, 10))
+
+    for checkpoint in checkpoints:
+
+        for page_type in checkpoints[checkpoint]:
+
+            # Save the number of different page types we have seen for the plot.
+            if page_type not in page_types:
+                page_types.append(page_type)
+
+            # As we may have multiple checkpoints, make sure we show the same page type on the same
+            # y value.
+            page_type_index = page_types.index(page_type)
+
+            ax.broken_barh(checkpoints[checkpoint][page_type], (page_type_index, bar_width),
+                           facecolors=f'tab:{colors[color_indx]}',
+                           label=f"{page_type} ({checkpoint})")
+
+            # Save the max offset seen for the plot assuming the offsets are sorted.
+            if checkpoints[checkpoint][page_type][-1][0] > max_addr:
+                max_addr = checkpoints[checkpoint][page_type][-1][0]
+                max_addr_size = checkpoints[checkpoint][page_type][-1][1]
+
+            # Change color for every page type and every checkpoint.
+            color_indx += 1
+
+    # Plot settings.
+    plt.yticks(np.arange(0, len(page_types) + 1, 1))
+    xlimit = max_addr + max_addr_size
+    plt.xlim(left=0,right=xlimit)
+    plt.xlabel("Offset")
+    plt.ylabel("Page type")
+    plt.legend()
+    plt.title(f"Distribution of each page type for {filename}")
+    plt.close()
+
+    # Generate the image.
+    img = mpld3.fig_to_html(fig)
+
+    return img
+
+def show_block_distribution_hist(filename, data, bins=100):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    """
+    for checkpoint in data:
+        all_addr = []
+
+        # Concatenate the lists for each page type:
+        for page_type in data[checkpoint]:
+            all_addr += data[checkpoint][page_type]
+
+    # Create a list of offsets, discarding the size component.
+    all_addr = list(map(itemgetter(0), all_addr))
+
+    # Plot settings.
+    fig, ax = plt.subplots(1, figsize=(15, 10))
+    ax.hist(all_addr, bins=bins)
+    plt.title(f"Distribution of blocks for {filename}")
+    plt.xlabel(f"Offset (bins={bins})")
+    plt.ylabel("Frequency")
```

Comment:
`Frequency (pages)`?

Reply:
As discussed, I am not sure what's the right word to put here. We usually talk about blocks when on-disk and each block correspond to a specific type of page.

Reply:
8d6a82d1d9e947cb81d2fb762964ba6876067782